toolset/benchmark/utils.py

```diff
@@ -138,3 +139,36 @@ def header(message, top='-', bottom='-'):
       else:
         result += "\n%s" % bottomheader
     return result + '\n'
+
+def check_services(services):
+
+  def check_service(address, port):
+    try:
+      s = socket.socket()
+      s.settimeout(5)
+      s.connect((address, port))
+      return (True, "")
+    except Exception as ex:
+      return (False, ex)
+    finally:
+      s.close
+
+  res = []
+  for s in services:
+    r = check_service(s[1], s[2])
+    res.append((s[0], r[0], str(r[1])))
+  return res
+
+def verify_database_connections(services):
+  allGo = True
+  for r in check_services(services):
+    if r[1]:
+      print r[0] + ": is GO!"
+    else:
+      print r[0] + ": is _NO_ GO: ERROR: " + r[2]
+      allGo = False
+
+  if allGo:
```

Comment:
My personal preference would be to avoid printing in utility methods if possible, and return a tuple `(allGo, errMessage)` variable to benchmarker.py who can then issue a print statement